### PR TITLE
DOC-4726 - POST `/{db}/_google` endpoint

### PR DESCRIPTION
**Pull request by jamesnocentini**
https://issues.couchbase.com/browse/DOC-4726

@bbrks This PR needs your review. I've basically copy-pasted the `/{db}/_facebook` endpoint and replaced `facebook` with `google` so there may well be other modifications to make.

In particular, I'm not sure whether the `google.register` property exists in the config? If so I'll update the config file too.

### CODE REVIEW

File: modules/ROOT/assets/attachments/sync-gateway-public.yaml [hunk 1]
```diff
@@ -1011,6 +1011,42 @@ paths:
...
+  /{db}/_google:
+    parameters:
+      - $ref: '#/parameters/db'
+    post:
+      tags:
+        - auth
+      summary: User creation / authentication via Google
+      description: |
+        Sync Gateway allows users to authenticate using a Google account. Your application is responsible for generating a Google token; this generally needs to be done by running the Google login flow inside a web-view and capturing the generated token. This endpoint can be used to check the validity of the access token.
+        To allow Google Login with Sync Gateway, it must be explicitely enabled in the Sync Gateway configuration file by setting the [google.register](config-properties.html#facebook-register) property to `true`.
```
Comment (bbrks) [inline modules/ROOT/assets/attachments/sync-gateway-public.yaml]:
```suggestion
        To allow Google Login with Sync Gateway, it must be explicitely enabled in the Sync Gateway configuration file by setting the [google.register](config-properties.html#google-register) property to `true`.
```

File: modules/ROOT/assets/attachments/sync-gateway-public.yaml [hunk 2]
```diff
@@ -1011,6 +1011,42 @@ paths:
...
+                  description: List of authentication methods.
+                  items:
+                    type: string
+                ok:
+                  type: boolean
+                  description: Always true if the operation was successful.
+                userCtx:
+                  $ref: '#/definitions/UserContext'
+            401:
+              description: Authentication failed. Facebook verification server status 400
```
Comment (bbrks) [inline modules/ROOT/assets/attachments/sync-gateway-public.yaml]:
```suggestion
              description: Authentication failed. Google verification server status 400
```

File: modules/ROOT/assets/attachments/sync-gateway-public.yaml [hunk 3]
```diff
@@ -1011,6 +1011,42 @@ paths:
...
+        Sync Gateway allows users to authenticate using a Google account. Your application is responsible for generating a Google token; this generally needs to be done by running the Google login flow inside a web-view and capturing the generated token. This endpoint can be used to check the validity of the access token.
+        To allow Google Login with Sync Gateway, it must be explicitely enabled in the Sync Gateway configuration file by setting the [google.register](config-properties.html#facebook-register) property to `true`.
+      parameters:
+        - in: body
+          name: body
+          description: Request body
+          schema:
+            type: object
+            properties:
+              access_token:
```
Comment (bbrks) [inline modules/ROOT/assets/attachments/sync-gateway-public.yaml]: For the Google login, this field is called `id_token`

```suggestion
              id_token:
```

File: modules/ROOT/assets/attachments/sync-gateway-public.yaml (hunk 1 above)
Comment (bbrks) [inline modules/ROOT/assets/attachments/sync-gateway-public.yaml]: SG config must also specify `app_client_id` in the config under Google.

**Review by bbrks — CHANGES_REQUESTED** [review]
Couple of changes that are different vs. the Facebook login endpoint.

File: modules/ROOT/assets/attachments/sync-gateway-public.yaml (hunk 1 above)
Comment (jamesnocentini) [reply modules/ROOT/assets/attachments/sync-gateway-public.yaml]: Addressed in https://github.com/couchbase/docs-sync-gateway/commit/5846a4b9797668b08e4936a0723a72e04f13b053

**Comment by jamesnocentini** [comment]
👍

**Pull request merged**